content/frontend/angular-apollo/2-queries-loading-links.md

```diff
@@ -0,0 +1,322 @@
+---
+title: "Queries: Loading link-items"
+pageTitle: "Fetching Data using GraphQL Queries with Angular & Apollo Tutorial"
+description: "Learn how you can use GraphQL queries with Apollo Client to load data from a server and display it in your Angular components."
+question: What's the idiomatic way for loading data with Angular & Apollo?
+answers: ["Using a higher-order component called 'graphql'", "Using the 'query' or 'watchQuery' method on Apollo service", "Using 'fetch' and putting the query in the body of the request", "Using XMLHTTPRequest and putting the query in the body of the request"]
+correctAnswer: 0
+---
+
+### Preparing the Angular components
+
+The first piece of functionality that you'll implement in the app is loading and displaying a list of `link-item` elements. You'll walk up our way in the Angular component hierarchy and start with the component that'll render a single link-item.
+
+<Instruction>
+
+Run the command below to generate all boilerplate files:
+
+```bash
+ng generate component link-item
+```
+
+</Instruction>
+
+<Instruction>
+
+This command will create a `link-item` folder containing several new files called `link-item.component.ts` , `link-item.component.spec.ts`, `link-item.component.html`, `link-item.component.css` in the `app` directory and add the following code in `link-item.component.ts`:
+
+```ts(path=".../hackernews-angular-apollo/src/app/link-item/link-item.component.ts")
+import {Component, Input, OnInit} from '@angular/core';
+import {Link} from '../types';
+
+@Component({
+  selector: 'hn-link-item',
+  templateUrl: './link-item.component.html',
+  styleUrls: ['./link-item.component.css']
+})
+export class LinkItemComponent implements OnInit {
+  @Input()
+  link: Link;
+
+  constructor() {
+  }
+
+  ngOnInit() {
+  }
+
+  voteForLink= async () => {
+    // ... you'll implement this in chapter 6
+  }
+}
+```
+
+```html(path=".../hackernews-angular-apollo/src/app/link-item/link-item.component.html")
+
+<div>{{link.description}} ({{link.url}})</div>
+
+```
+</Instruction>
+
+Note, we will be writing all our typings in a `./src/app/types.ts` file and simply importing these types into components as needed.
+
+<Instruction>
+
+This command will create a `link-item` folder containing several new files called `link-item.component.ts` , `link-item.component.spec.ts`, `link-item.component.html`, `link-item.component.css` in the `app` directory and add the following code in `link-item.component.ts`:
+
+```ts(path=".../hackernews-angular-apollo/src/app/type.ts")
+export class Link {
+  id: string;
+  description: string;
+  url: string;
+  createdAt: string;
+}
+
+```
+</Instruction>
+
+
+This is a simple Angular component that expects a `link` in its `Input` (`link-item.component.ts`) and renders the link's `description` and `url` (`link-item.component.html`). Easy as pie! 🍰
+
+Next, you'll implement the component that renders a list of link-items.
+
+<Instruction>
```

Comment:
this instruction block doesn't render correctly. that's because our markdown parser is a bit limited and has the rule that one instruction block can at most contain one code block (this one contains two code blocks) and that code block also needs to be that _last_ element inside the instruction block.

Reply:
the quick workaround to just make 2 instructions blocks out of this one